pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -102,7 +111,12 @@ def _configure_model_summary_callback(self, weights_summary: Optional[str] = Non
                     f" but got {weights_summary}",
                 )
             max_depth = ModelSummaryMode.get_max_depth(weights_summary)
-            model_summary = ModelSummary(max_depth=max_depth)
+            if self.trainer._progress_bar_callback is not None and isinstance(
```

Comment:
@kaushikb11  don't we want to use rich summary when rich is installed automatically?

Reply:
@awaelchli we ran into one issue where we had an old rich version installed (8.x instead of 10.x), but the `_RICH_AVAILABLE` check defaulted to using the rich progress bar because the package was available. 

If we default to using Rich, let's make sure the package versioning requirements for Rich are also part of the availability check

Reply:
There's still a discussion going on whether we should add it to Bolts or not. Would wait for that.

Reply:
I'll make an issue for this, and we can enable this in a separate PR :)